internal/service/pushbasedlogexport/state_transition.go

```diff
@@ -0,0 +1,77 @@
+package pushbasedlogexport
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/retrystrategy"
+)
+
+const (
+	ActiveState         = "ACTIVE"
+	UnconfiguredState   = "UNCONFIGURED"
+	InitiatingState     = "INITIATING"
+	BucketVerifiedState = "BUCKET_VERIFIED"
+)
+
+func WaitStateTransition(ctx context.Context, projectID string, client admin.PushBasedLogExportApi,
+	timeConfig retrystrategy.TimeConfig) (*admin.PushBasedLogExportProject, error) {
+	stateConf := &retry.StateChangeConf{
+		Pending:    []string{InitiatingState, BucketVerifiedState},
+		Target:     []string{ActiveState},
+		Refresh:    refreshFunc(ctx, projectID, client),
+		Timeout:    timeConfig.Timeout,
+		MinTimeout: timeConfig.MinTimeout,
+		Delay:      timeConfig.Delay,
+	}
+
+	result, err := stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if logConfig, ok := result.(*admin.PushBasedLogExportProject); ok && logConfig != nil {
+		return logConfig, nil
+	}
+	return nil, errors.New("did not obtain valid result when waiting for push-based log export configuration state transition")
+}
+
+func WaitResourceDelete(ctx context.Context, projectID string, client admin.PushBasedLogExportApi, timeConfig retrystrategy.TimeConfig) error {
```

Comment:
knit: don't know if it's possible to have only one wait function or they're different enough to have 2

Reply:
They are different in terms of target and pending states